### PR TITLE
Cleanup around testing and packaging lifecycle

### PR DESCRIPTION
**Pull request by psafont**
I've simplified managing with python packaging, removing test-requirements.txt and removing the direct usage of setup.py for installing the package and dependencies: that can be handled through pip and PEP518.

I also think there is a case to remove requirements.txt as it is, since it does not list all the dependencies needed in the environment, nor they are locked.

Let me know your thoughts, especially about requirements.txt. I haven't deleted the file but there is no other piece of code that actually uses it.

### CODE REVIEW

**Comment by psafont** [comment]
Python 3.4 is failing on appveyor because the wheel for 5.4.8 is not uploaded to PyPI: the compilation fails: https://github.com/giampaolo/psutil/issues/1356

**Comment by psafont** [comment]
For Travis the failure appeared when enabling [isolated builds](https://tox.readthedocs.io/en/latest/config.html#conf-isolated_build) (PEP518)

I've reported the issue to tox-travis, to try to see where the issue is really: https://github.com/tox-dev/tox-travis/issues/119

File: tox.ini [hunk 1]
```diff
@@ -7,39 +7,24 @@ envlist =
...
-  3.5: py35
-  3.6: py36
-  3.7: py37
-
 [testenv]
-passenv =
-  CI
-  TRAVIS
-  TRAVIS_*
+passenv = CI TRAVIS*
```
Comment (ryanhiebert) [inline tox.ini]: This shouldn't be necessary to make tox-travis work. I don't think it should cause issues either, though.

**Comment by mr-c** [comment]
The travis failure is a manifestation of https://github.com/common-workflow-language/cwltool/issues/954

File: tox.ini (hunk 1 above)
Comment (psafont) [reply tox.ini]: It's necessary to be able to upload coverage reports to codecov, though

**Comment by psafont** [comment]
I need to make sure the coverage reporting still works fine. After that the only issue there is is the (unrelated?) Travis failure.


After that is fixed there are a some thing that might be worth doing, maybe for another PR:
- Activate isolated builds for tox when setuptools releases a version fixing the setup.py bug.
- Clean jenkins' scripts, I'm not sure why `coverage` is used the way it's used, among other things.
- Clean Makefile: is all that's there actually used?
- Use [this script](https://gist.github.com/eli-collins/ef91add2bc635ccaf61738cd805cb18b) to filter which environments are run on appveyor, otherwise tox detects all the python versions installed and executes all tests.

**Comment by psafont** [comment]
Coverage reporting seems to work find according to https://codecov.io/gh/common-workflow-language/cwltool/pull/961?src=pr&el=desc (80.43%, +2.23%)

**Comment by mr-c** [comment]
Would including `setup.py` in MANIFEST.in be sufficient to fix the setuptools bug?

**Comment by psafont** [comment]
I also was thinking if we would want to delete of `release-test.sh`: the test of building the package in an  isolated virtual environment and installing it in another is already being done with tox :)

File: Makefile [hunk 2]
```diff
@@ -75,10 +67,8 @@ dist/${MODULE}-$(VERSION).tar.gz: $(SOURCES)
 
 ## clean       : clean up all temporary / machine-generated files
 clean: FORCE
-	rm -f ${MODILE}/*.pyc tests/*.pyc
+	git clean -ffdx || true
```
Comment (mr-c) [inline Makefile]: No, this will delete too much . `git clean -fdx` is appropriate for `distclean` but not `clean`

**Comment by psafont** [comment]
Code coverage on codecov seems to be unaffected by parallellizing the tests, compared to https://github.com/common-workflow-language/cwltool/pull/961#issuecomment-435844827

**Pull request closed without merging**